oap-server/server-storage-plugin/storage-elasticsearch-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/elasticsearch/query/TraceQueryEsDAO.java

```diff
@@ -125,7 +127,9 @@ public TraceBrief queryBasicTraces(long startSecondTB, long endSecondTB, long mi
         sourceBuilder.query(QueryBuilders.termQuery(SegmentRecord.TRACE_ID, traceId));
         sourceBuilder.size(20);
 
-        SearchResponse response = getClient().search(SegmentRecord.INDEX_NAME, sourceBuilder);
+        ElasticSearchClient client = getClient();
+        String[] indexes = client.getIndexNameByDate(SegmentRecord.INDEX_NAME, 0, 0);
```

Comment:
`0, 0` is not a friendly way to say all indexes. You should redesign the API to have the index name list.